### PR TITLE
refactor to use sg-bucket. 

### PR DESCRIPTION
**Pull request by tleyden**
I've tested that the following works on a clean machine (via CI server):

```
$ git clone git://github.com/tleyden/walrus.git /home/ubuntu/src/github.com/tleyden/walrus 
Cloning into '/home/ubuntu/src/github.com/tleyden/walrus'...
$ go version
go version go1.4.2 linux/amd64
$ go get -t
$ go build
$ go test -short
```

### CODE REVIEW

**Comment by tleyden** [comment]
@adamcfraser btw I removed the travis.yml since I don't think travis is being used for this repo (there was no badge or anything)  Please ignore the fail checks

File: crud.go [hunk 1]
```diff
@@ -51,8 +54,8 @@ type lolrusDoc struct {
 
 // Creates a simple in-memory Bucket, suitable only for amusement purposes & testing.
 // The Bucket is created empty. There is no way to save it persistently.
-func NewBucket(bucketName string) Bucket {
-	ohai("NewBucket %s", bucketName)
+func NewBucket(bucketName string) sgbucket.Bucket {
+	log.Printf("NewBucket %s", bucketName)
```
Comment (adamcfraser) [inline crud.go]: It looks like we're losing the conditional logging and "Walrus:" prefixing from ohai - I think we'd want to keep both of those.

Comment (tleyden) [inline crud.go]: Good catch, thanks.  I will fix that.

**Comment by adamcfraser** [comment]
Looks good.

**Pull request merged**